### PR TITLE
Detector changes for run image extension

### PR DESCRIPTION
**Pull request by natalieparellano**
Fixes https://github.com/buildpacks/lifecycle/issues/996

### CODE REVIEW

File: acceptance/detector_test.go [hunk 1]
```diff
@@ -362,11 +362,13 @@ fail: fail_detect_buildpack@some_version
 					"-extensions=/cnb/extensions",
 					"-generated=/layers/generated",
 					"-log-level=debug",
+					"-run=/layers/run.toml", // /cnb/run.toml is the default location of run.toml
 				),
 			)
 
 			t.Log("runs /bin/detect for buildpacks and extensions")
 			h.AssertStringContains(t, output, "Platform requested experimental feature 'Dockerfiles'")
+			h.AssertStringContains(t, output, "FOO=val-from-build-config")
```
Comment (natalieparellano) [inline acceptance/detector_test.go:371]: Unrelated test

**Comment by natalieparellano** [comment]
This is ready for feedback but we can block merging on https://github.com/buildpacks/lifecycle/pull/994

**Comment by natalieparellano** [comment]
The exporter acceptance failure is mystifying me...

File: cmd/lifecycle/detector.go [hunk 2]
```diff
@@ -18,28 +18,24 @@ type detectCmd struct {
...
-		cli.FlagPlatformDir(&d.PlatformDir)
-	default:
-		cli.FlagAppDir(&d.AppDir)
-		cli.FlagBuildpacksDir(&d.BuildpacksDir)
-		cli.FlagGroupPath(&d.GroupPath)
-		cli.FlagOrderPath(&d.OrderPath)
-		cli.FlagLayersDir(&d.LayersDir)
-		cli.FlagPlanPath(&d.PlanPath)
-		cli.FlagPlatformDir(&d.PlatformDir)
-		cli.FlagBuildConfigDir(&d.BuildConfigDir)
```
Comment (natalieparellano) [inline cmd/lifecycle/detector.go:41]: Unrelated bug, this should only be valid for newer platforms

File: platform/files.go [hunk 3]
```diff
@@ -49,6 +49,43 @@ type ImageIdentifier struct {
...
+				found = true
+				continue
+			}
+		}
+		if !found {
+			return false
+		}
+	}
+	return true
+}
```
Comment (natalieparellano) [inline platform/files.go:87]: Copied from further down in the file

File: generator.go [hunk 4]
```diff
@@ -123,24 +148,50 @@ func (g *Generator) Generate() (GenerateResult, error) {
...
-	g.Logger.Debugf("Finished build, selected runImage '%s'", runImage)
-	return GenerateResult{Plan: filteredPlan, UsePlan: true, RunImage: runImage}, nil
+	return GenerateResult{
+		AnalyzedMD: newAnalyzedMD,
+		Plan:       filteredPlan,
+		UsePlan:    true,
+	}, nil
+}
+
+func containsMatch(images []platform.RunImageForExport, imageName string) bool {
```
Comment (joe-kimmel-vmw) [inline generator.go:184]: nit: given that len 0 is a wildcard match, is there a different word than "contains" ?  eg. `isMatch` ? `matches`? `satisfies`?

File: buildpack/dockerfile.go [hunk 5]
```diff
@@ -64,80 +82,113 @@ func VerifyBuildDockerfile(dockerfile string, logger log.Logger) error {
...
+	if stages[0].BaseName != baseImageArgRef {
+		return errors.New(errBuildMissingRequiredFROMCommand)
 	}
 
 	return nil
 }
 
-func VerifyRunDockerfile(dockerfile string) error {
+func VerifyRunDockerfile(dockerfile string, buildpackAPI *api.Version, logger log.Logger) (string, error) {
+	if buildpackAPI.LessThan("0.10") {
```
Comment (jabrown85) [inline buildpack/dockerfile.go:124]: I'm surprised we are keeping this branching logic - do we have such rules for experimental features?

**Review by jabrown85 — APPROVED** [review]

File: buildpack/dockerfile.go (hunk 5 above)
Comment (natalieparellano) [reply buildpack/dockerfile.go:124]: Yeah... maybe it's a bit overzealous. I can remove it

Comment (natalieparellano) [reply buildpack/dockerfile.go]: Updated in 34e396a3

File: generator.go (hunk 4 above)
Comment (natalieparellano) [reply generator.go]: Updated in b1c2c2e

**Pull request merged**